src/Quasar/QuasarF/Interpreter/Internal.purs

```diff
@@ -149,32 +151,50 @@ mkRequest
   . (String → Either Error a)
   → AXF.AffjaxF RequestContent String
   → Free (Coproduct l AXFP) (Either QError a)
-mkRequest f = map (handleResult f) <<< liftF <<< right
+mkRequest f a =
+  map (map snd) $ mkRequest' f a
+
+mkRequest'
+  ∷ ∀ a l
+  . (String → Either Error a)
+  → AXF.AffjaxF RequestContent String
+  → Free (Coproduct l AXFP) (Either QError (Tuple (AX.AffjaxResponse String) a))
+mkRequest' f = map (handleResult f) <<< liftF <<< right
+
+withExpired ∷ ∀ a b. Tuple (AX.AffjaxResponse a) b → ExpiredContent b
+withExpired (Tuple { headers: hs } content) =
+  ExpiredContent
+    { content
+    , expired: Array.any isExpired hs
+    }
+  where
+    -- TODO: Vince, I really need to make this look beter and work out how to make tests ¯\_(ツ)_/¯
```

Comment:
you can write `isHeadersExpired :: Array RequestHeader -> Boolean` (or something) as top level function 
and test it with some predefined set of headers for desired outputs